### ISSUE TITLE
Cake.CoreCLR can't handle whitespace in path

### ISSUE THREAD

**Issue by Roadrunner67** [cake-build/cake#2507]
I know about #2214, and that seems to have been solved, but there is more to it:

Steps to reproduce:

```
mkdir 'My Folder'
pushd 'My Folder'
nuget install Cake.CoreCLR -o tools
dotnet new console -n HelloWorld
dotnet build .\HelloWorld\
echo 'DotNetCoreExecute("./HelloWorld/bin/Debug/netcoreapp2.2/Helloworld.dll");' > build.cake
dotnet .\tools\Cake.CoreCLR.0.32.1\Cake.dll
popd

```
### What You Are Seeing?
No executable found matching command "dotnet-C:/Temp/My"
Error: One or more errors occurred. (.NET Core CLI: Process returned an error (exit code 1).)
        .NET Core CLI: Process returned an error (exit code 1).

### What is Expected?
Hello World!

### What version of Cake are you using?
0.32.1

### Are you running on a 32 or 64 bit system?
64

### What environment are you running on?  Windows? Linux? Mac?
Windows

If 'My Folder' is changed to 'MyFolder', everything works as expected.

**Comment by Roadrunner67** [comment]
The problem seems to be in DotNetCoreExecutor.GetArguments(...)
where:
...
            assemblyPath = assemblyPath.IsRelative ? assemblyPath.MakeAbsolute(_environment) : assemblyPath;
            builder.Append(assemblyPath.FullPath);
...
should be:
...
            assemblyPath = assemblyPath.IsRelative ? assemblyPath.MakeAbsolute(_environment) : assemblyPath;
            builder.Append(AddQuotesIfRequired(assemblyPath.FullPath));
... 

And AddQuotesIfRequired could be something like:

        private string AddQuotesIfRequired(string path)
        {
            return !string.IsNullOrWhiteSpace(path) ?
                path.Contains(" ") && (!path.StartsWith("\"") && !path.EndsWith("\"")) ?
                    "\"" + path + "\"" : path :
                    string.Empty;
        }

This goes IMHO for every use of assemblyPath.MakeAbsolute.

**Comment by Roadrunner67** [comment]
The PR failed during tests against other OS'es than Windows.
I think modifying the tests to expect quoted paths will also benefit the other OS'es, but I might not be the one to judge.

**Comment by mholo65** [comment]
@Roadrunner67 I think qouting paths is generally a good idea, we do this in other places as well! I see tests are now passing on other OS’s in your PR.

**Comment by devlead** [comment]
fixed by #2511
